spec/features/recipes/show_spec.rb

```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+RSpec.describe "recipes show page" do
+  let!(:recipe) { Recipe.create!(name: "Garlic Beans", complexity: 1, genre: "Legumes")}
+  let!(:ingredient_1) { recipe.ingredients.create!(name: "Garlic Powder", cost: 1)}
+  let!(:ingredient_2) { recipe.ingredients.create!(name: "Beans", cost: 3)}
+
+  it "displays a specific recipes name, complexity, genre and list of ingredients" do
+    visit "/recipes/#{recipe.id}"
+
+    expect(page).to have_content("#{recipe.name}")
+    expect(page).to have_content("Complexity: #{recipe.complexity}")
+    expect(page).to have_content("Genre: #{recipe.genre}")
+    expect(page).to have_content("Ingredients:")
```

Comment:
This section would be a perfect use case for a `within` block - in fact anywhere you have a `.each` in your views, you could benefit from a within block because all that data to be displayed will be dynamic.